dsl/pace/dsl/stencil.py

```diff
@@ -365,6 +426,10 @@ def __call__(self, *args, **kwargs) -> None:
         if self.stencil_object is None:
             self.stencil_object = self._compile()
 
+        exec_info = self.timing_collector.exec_info.setdefault(
+            _stencil_object_name(self.stencil_object), {}
+        )
```

Comment:
You can avoid this setdefault if you initialize exec_info as a `collections.defaultdict(dict)`. You can only do this if you make exec_info private (i.e. not possible to pass as an init argument of TimingCollector) to guarantee it's a defaultdict and not some other dict.

Reply:
Hmm, I tried that but I think I prefer this solution because it allows `exec_info` to remain public.

Reply:
you can still make it readable-public with a property method that returns `self._exec_info`, it's just that you can't have it as an init argument. I don't see any use case for passing it as an init argument, anyways.